src/Analyzer/Resolve/QueryAnalyzer.cpp

```diff
@@ -3766,7 +3766,10 @@ ProjectionNames QueryAnalyzer::resolveExpressionNode(
 
                 auto hints = TypoCorrection::collectIdentifierTypoHints(unresolved_identifier, valid_identifiers);
 
-                throw Exception(ErrorCodes::UNKNOWN_IDENTIFIER, "Unknown {}{} identifier {} in scope {}{}",
+                throw Exception(
+                    scope.context->getInitialQueryId(),
```

Comment:
There were so many things to discuss about requirements, I decided it's much easier to just raise PR and discuss it here. This query_id above is just an example of how this could be used and also allows me to test. If this change is what you wanted to build - I can go ahead and add this id to all exceptions that have access to query_id.

Reply:
>There were so many things to discuss about requirements, I decided it's much easier to just raise PR and discuss it here

Yep, this is always easier

As for the implementation - let's simply go this way - https://github.com/ClickHouse/ClickHouse/pull/76581#discussion_r1965387562

That way you will not need to modify any single place of `Exception`